pkg/applier/applier_test.go

```diff
@@ -139,23 +135,6 @@ func TestApply(t *testing.T) {
 				formApplyEvent(event.ApplyPending, testObj2, nil),
 			},
 			expectedError: ErrorForResourceWithResource(errors.New("unknown type"), idFrom(testID), testObj),
-			expectedCSEs: []v1beta1.ConfigSyncError{
```

Comment:
Does this still verify the `resources` fields? I don't think this is all surfaced by the `error` interface, so it seems like this might be tested - although I haven't yet dug into the `testutils` code from cli-utils

/hold

Reply:
This also reduces the test to basically asserting `foo() == foo()`, where `foo` is the real code path. The test assertions should be written such that `foo() == expectedValue`. Reading this test it's not really clear what the output should look like

Reply:
The test here is asserting that Applier.Apply returns the error received from a specific event and wraps it appropriately with the resource.

Reply:
With this test I could change the underlying implementation of `ErrorForResourceWithResource` to no longer include the ResourceRef and the test would still pass

Reply:
Yes, but this test isn't testing that ErrorForResourceWithResource exposes the right fields/methods. It's just testing that Applier.Apply returned the expected error. 

If you want to confirm ErrorForResourceWithResource has a ResourceRef field/method, that should be a test of ErrorForResourceWithResource itself.

Reply:
Can you add some individual tests for that then, since the coverage is being removed here?

Reply:
Fine, but we have zero tests for error builders and dozens of usages of BuildWithResources already. ErrorForResourceWithResource just wraps BuildWithResources the same way everything else does.

Reply:
Agreed, which is why I asked for these assertions to be added to these tests in the first place. I find the error builders themselves pretty obfuscated and untested.

This bugfix was specifically for the ResourceRef fields on the RSync status, so I wanted to make sure there were assertions in place which verify the behavior and prevent regression

Reply:
Done

Reply:
Thank you!